Sources/BraintreeThreeDSecure/Public/BraintreeThreeDSecure/BTThreeDSecureRequest.h

```diff
@@ -140,6 +154,15 @@ typedef NS_ENUM(NSInteger, BTThreeDSecureShippingMethod) {
  */
 @property (nonatomic) BOOL dataOnlyRequested;
 
+/**
+ Optional. An authentication created using this property should only be used for adding a payment method to the merchant's vault and not for creating transactions.
+ Defaults to BTThreeDSecureAddCardChallengeUnspecified.
+ If set to BTThreeDSecureAddCardChallengeRequested, the authentication challenge will be requested from the issuer to confirm adding new card to the merchant's vault.
+ If set to BTThreeDSecureAddCardChallengeUnspecified and amount is 0, the authentication challenge will be presented to the user.
+ If set to BTThreeDSecureAddCardChallengeNotRequested, when the amount is 0, the authentication challenge will not be presented to the user
+ */
+@property (nonatomic, assign) BTThreeDSecureCardAddChallenge cardAddChallengeRequested;
```

Comment:
Since this is an enum now, maybe the name `cardAddChallengeRequested` isn't the most clear. 

So instead of merchant code looking like:
`request.cardAddChallengeRequested = .requested`

maybe we can do something like
`request.cardAddChallengeType = .requested` or `request.cardAddChallenge = .requested` or `request.cardAddChallengeRequirement = . requested` ? Not sure on which would be best, but we can brainstorm. Thoughts?

Reply:
Also, for the `notRequested` option the docstring says that the challenge will _not_ be presented, in any case. Am I understanding that right?

If that's the case - I feel like the name `notRequested` might not communicate that. 🤔

Reply:
After reviewing the updated JS doc strings and GW PR on this I'm thinking we might want to reword the doc strings here. I don't know that we can say for sure that a challenge will be presented to the user, just that an add card authentication will be requested from the issuer.

The breakdown is: 
true -> card add challenge will be requested
false -> card add challenge will not be requested
null and amount 0 -> card add challenge will be requested
null and amount greater than 0 -> card add challenge will not be requested

Reply:
`null and amount 0 -> card add authentication will be requested` Blade confirmed was behavior before this PR, as well so we are good on that not being a breaking change.

Reply:
Yeah I think we are good on the breaking change part 👍 . I've updated the param to be set as `request.cardAddChallenge`, and updated the doc string to try and provide more clarity on the different scenarios.

Since this param is actually determining whether or not the card add authentication will be requested from the issuer (rather than what specifically will be displayed to the user), I think `.requested` and `.notRequested` still make the most sense here, but open to other ideas.

Reply:
Awesome - these clarifications are 👌  super helpful